.github/workflows/docker-image.yml

```diff
@@ -0,0 +1,36 @@
+name: Docker Image CI
+
+on:
+  push:
+    branches:
+      - '**'
+    tags:
+      - 'libp2p-server-**'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ github.token }}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Get branch or tag name
+        id: ref-name
+        run: echo ::set-output name=ref::${GITHUB_REF#refs/*/}
```

Comment:
I think there is a GitHub action from docker for extracting useful tags, might be useful here!

Reply:
True, there is `docker/metadata-action@v3`. That said, for such a simple problem, do we really need to import an external tool?

Reply:
This solution doesn't seem to be very robust in the case of more complex branch names: https://github.com/libp2p/rust-libp2p/actions/runs/5961028379/job/16169455319?pr=4386#step:6:113

Reply:
Good call-out. See https://github.com/libp2p/rust-libp2p/pull/4421.